cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/LoadPersistedConcepts.kt

```diff
@@ -0,0 +1,318 @@
+    private data class ConceptRootEntry(
+        val concept: ConceptEntry,
+        val location: LocationEntry?,
+        val signature: SignatureEntry?,
+    )
+
+    /**
+     * This class represents a single concept entry in the YAML/JSON file. It contains the name of
+     * the concept, optional constructor arguments, and optional DFG connections.
+     *
+     * @param name The FQN of the concept to be added.
+     * @param constructorArguments The constructor arguments to be passed to the concepts'
+     *   constructor.
+     * @param dfg The DFG connections to be created between the concept and the underlying node.
+     */
+    private data class ConceptEntry(
+        val name: String,
+        val constructorArguments: List<ConstructorArgumentEntry>?,
+        val dfg: DFGEntry?,
+    )
+
+    /**
+     * This class represents the DFG connections to be created between the concept and the
+     * underlying node.
+     *
+     * @param fromThisNodeToConcept Whether to add a DFG connection from the underlying node to the
+     *   concept node.
+     * @param fromConceptToThisNode Whether to add a DFG connection from the concept node to the
+     *   underlying node.
+     */
+    private data class DFGEntry(
+        val fromThisNodeToConcept: Boolean?,
+        val fromConceptToThisNode: Boolean?,
+    )
+
+    /**
+     * This class represents a single constructor argument entry in the YAML/JSON file.
+     *
+     * @param name The name of the constructor argument.
+     * @param value The value of the constructor argument.
+     */
+    private data class ConstructorArgumentEntry(val name: String, val value: String)
+
+    /**
+     * This class represents a single location entry in the YAML/JSON file.
+     *
+     * @param file The file this entry applies to. E.g. `file:/foo/bar/baz/concepts.yaml`.
+     * @param region The region within the [file]. E.g. `1:1-2:2`.
+     * @param type Optionally, the type of the node to match against. E.g.
+     *   `de.fraunhofer.aisec.cpg.graph.statements.expressions.CallExpression`.
+     */
+    private data class LocationEntry(val file: String, val region: String, val type: String?)
+
+    /**
+     * This class represents a single signature entry in the YAML/JSON file.
+     *
+     * @param fqn The fully qualified name of the [CallExpression] to match against. E.g.
+     *   `foo.bar.baz`.
+     */
+    private data class SignatureEntry(
+        val fqn: String
+        // Extend this class with arguments / types as needed in the future
+    )
+
+    /**
+     * This function retrieves the [Node.calls] from the [TranslationResult] by their fully
+     * qualified name (FQN). The match is performed on the [CallExpression.reconstructedImportName].
+     *
+     * @param fqn The fully qualified name of the calls to retrieve.
+     * @return A list of [CallExpression] nodes matching the provided FQN.
+     */
+    private fun TranslationResult.getCallsByFQN(fqn: String): List<Node> {
```

Comment:
This should be part of the `Extensions` as it looks like a useful functionality in many cases. I would have assumed that this should even be exposed as a shortcut as part of the `calls(...)` and `calls[...]` behavior but probably it got lost with recent changes of how the name is created.